rust/gui-client/src-tauri/src/client/ipc.rs

```diff
@@ -93,18 +77,44 @@ impl Client {
             }
             Ok(())
         });
+        Ok(Self { task, tx })
+    }
 
-        let mut client = Self { task, tx };
-        let token = token.expose_secret().clone();
-        client
-            .send_msg(&IpcClientMsg::Connect {
-                api_url: api_url.to_string(),
-                token,
-            })
+    pub(crate) async fn disconnect_from_ipc(mut self) -> Result<()> {
+        self.task.abort();
+        self.tx.close().await?;
+        Ok(())
+    }
+
+    pub(crate) async fn disconnect_from_firezone(&mut self) -> Result<()> {
+        self.send_msg(&IpcClientMsg::Disconnect)
+            .await
+            .context("Couldn't send Disconnect")?;
```

Comment:
Is it worth for this function to return an error? What is the caller going to do with that?

I'd suggest we just log something in here instead and make the API a bit more ergonomic.

Reply:
🤔 The only error that could happen is that the IPC connection could have dropped, in which case the GUI won't work anymore.

And the GUI has no logic to reconnect to the IPC service, so not only should we log, we should probably just crash if this happens.

How about this:
- Inline `connect_to_firezone`, `disconnect_from_firezone`, `reconnect`, and `set_dns` since they're just wrappers around `send_msg` with one call site each.
- Now that the GUI knows that any error from `send_msg` is an IPC failure, if we get that error, just log it, show an error dialog, and then crash. We don't have any recovery plan and the user should know that Firezone isn't going to work until it's restarted.

Reply:
Sounds good to me!

Reply:
> * show an error dialog

We may want to reconsider this. I'd suggest a regular notification instead as this may happen with FZ being the active window and an alert / dialog forces itself into the foreground.

It would be annoying to trigger a pop-up alert if the admin e.g. kills the IPC service on purpose.